presto/presto.go

```diff
@@ -416,6 +420,14 @@ func (c *Conn) roundTrip(ctx context.Context, req *http.Request) (*http.Response
 			}
 			switch resp.StatusCode {
 			case http.StatusOK:
+				catalog := resp.Header.Get(prestoSetCatalogHeader)
+				if catalog != "" {
+					c.httpHeaders.Set(prestoCatalogHeader, catalog)
+				}
+				schema := resp.Header.Get(prestoSetSchemaHeader)
```

Comment:
should we handle `X-Set-Path` here too?

Reply:
I'm not sure. It's not set in the initial connection. In what cases the server would send it and expect it to be set in future requests?

Reply:
LMK if you think we should add this for consistency.

Reply:
Actually there are more of those:

https://github.com/trinodb/trino/blob/83bcdffc4c8dd0ce377c5f12609c69b9088ba6a0/presto-client/src/main/java/io/prestosql/client/PrestoHeaders.java#L27-L32

I think we do not have to support all of those initially. But we should at least notice that they are present in the response and return an error to the caller, if we get an unsupported `X-Prest-Set-...` header. Otherwise we will pretend that we support `SET SESSION ...` syntax, but we would not send set session properties in followup queries.

Could you please add validation and test coverage?

Reply:
Done! Added this as a separate commit.